### PR TITLE
deployment: generate modules.yaml

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

**Comment by matz-e** [comment]
It's not really perfect, but it should be usable and better than nothing. Only whitelists all the software that we ever allow as modules, so `spack load` should be somewhat consistent (not that I advise using it).

File: deploy/modules.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    end
+  end
+end
+
+dumped = false
+File.foreach('configs/applications/modules.yaml') do |line|
+    if line =~ /^\s*whitelist:\s*$/../^\s*blacklist:\s*$/
+      if /^(?<prefix>\s*-\s*)(?<quote>['"]?)(?<pkg>.*)\k<quote>$/ =~ line
+        if not dumped
+          environ.sort.each do |pkg|
```
Comment (ohm314) [inline deploy/modules.rb:21]: name clashing

**Review by ohm314 — CHANGES_REQUESTED** [review]
almost...

**Review by ohm314 — CHANGES_REQUESTED** [review]
lgtm

**Review by ohm314 — APPROVED** [review]

**Pull request merged**